Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/model/CoreDevice.kt

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.features.input.model
+
+import androidx.annotation.Keep
+
+/**
+ * Represents a C++ ciface::Core::Device.
+ */
+class CoreDevice @Keep private constructor(@Keep private val pointer: Long) {
+    /**
+     * Represents a C++ ciface::Core::Device::Control.
+     *
+     * This class is non-static to ensure that the CoreDevice parent does not get garbage collected
```

Comment:
Nothing is static in kotlin (except when using `@JvmStatic` for Java interop) and `inner` just allows the class to use references to members of the outer class and makes the class only accessible via a reference to the outer class. But because of that, would it be appropriate to remove that part of the comment?

Reply:
If `inner` lets the class reference the outer class, you should go with my suggestion. The reference to the outer class is what prevents the outer class from being garbage collected.